website/static/js/pages/base-page.js

```diff
@@ -92,6 +93,100 @@ $(document).on('click', '.panel-heading', function(){
     }
 });
 
+//choose which emails/users to add to an account and which to deny
+function confirmEmails(emailsToAdd) {
+    if (emailsToAdd.length > 0) {
+        var confirmedEmailURL = '/api/v1/confirmed_emails/';
+        var email = emailsToAdd[0];
+        var title;
+        var requestMessage;
+        var confirmMessage;
+        var nopeMessage;
+        if (email.user_merge) {
+            title = 'Merge account';
+            requestMessage = 'Would you like to merge \<b>' + email.address + '\</b> into your account?  ' +
+                'This action is irreversable.';
+            confirmMessage = '\<b>' + email.address + '\</b> has been merged into your account.';
+            nopeMessage = 'You have chosen to not merge \<b>' + email.address + '\</b>  into your account. ' +
+                'If you change your mind, visit the \<a href="/settings/account/">user settings page</a>.';
+        }
+        else {
+            title = 'Add email';
+            requestMessage = 'Would you like to add \<b>' + email.address + '\</b> to your account?';
+            confirmMessage = '\<b>' + email.address + '\</b> has been added into your account.';
+            nopeMessage = 'You have chosen not to add \<b>' + email.address + '\</b> to your account. ' +
+                'If you change your mind, visit the \<a href="/settings/account/">user settings page</a>.';
+        }
+
+        var confirmFailMessage = 'There was a problem adding \<b>' + email.address +
+            '\</b>. Please contact <a href="mailto: support@osf.io">support@osf.io</a> if the problem persists.';
+
+        var cancelFailMessage = 'There was a problem removing \<b>' + email.address +
+            '\</b>. Please contact <a href="mailto: support@osf.io">support@osf.io</a> if the problem persists.';
+
+        bootbox.dialog({
+            title: title,
+            message: requestMessage,
+            onEscape: function () {
+            },
+            backdrop: true,
+            closeButton: true,
+            buttons: {
+                cancel: {
+                    label: 'Do not add email',
+                    className: 'btn-default',
+                    callback: function () {
+                        $.ajax({
+                            type: 'delete',
+                            url: confirmedEmailURL,
+                            contentType: 'application/json',
+                            dataType: 'json',
+                            data: JSON.stringify(email)
+                        }).done(function () {
+                            $osf.growl('Warning', nopeMessage, 'warning', 8000);
```

Comment:
Minor: Is this necessary? Showing a growl after a "cancel" operation seems superfluous.